pkg/generator/schema_generator.go

```diff
@@ -92,11 +91,6 @@ func (g *schemaGenerator) generateReferencedType(ref string) (codegen.Type, erro
 	sg := g
 
 	if fileName != "" {
-		fileExt := filepath.Ext(fileName)
-		if fileExt != ".json" {
-			return nil, fmt.Errorf("%w: '%s', only json schema files are allowed", ErrUnsupportedRefExtension, fileExt)
-		}
```

Comment:
The call to `g.fileLoader.Load` is already checking for supported references https://github.com/omissis/go-jsonschema/blob/main/pkg/schemas/loaders.go#L143-L149, hence it seems unnecessary to perform this check here again.